src/pnm.imageio/pnminput.cpp

```diff
@@ -339,7 +348,13 @@ PNMInput::open(const std::string& name, ImageSpec& newspec,
                const ImageSpec& config)
 {
     ioproxy_retrieve_from_config(config);
-    return open(name, newspec);
+    if (!open(name, newspec)) {
+        errorfmt("Could parse spec for file \"%s\"", name);
+        return false;
+    }
+
+    m_spec["pnm:pfmflip"] = config.get_int_attribute("pnm:pfmflip", 1);
```

Comment:
`m_spec` is for communicating what's in the file to the caller.

`config` is for the caller to give us instructions about how to read.

So it's unusual to copy something from the config into m_spec. I suspect you are merely trying to communicate that config hint to the later reading functions, but usually what we do in a case like this is just have a `bool` as a private member of the PNMInput, set it here, and use it as needed in the read_scanline function.

Reply:
I think at the very least, we should only set this in m_spec if it really is a pfm file, not for the non-float varieties.